### PR TITLE
exercise/highcharts-9-part1

### PR DESCRIPTION
**Pull request by hubertkozik**

### CODE REVIEW

**Review by Izothep — CHANGES_REQUESTED** [review]
In general ok, works well. What about using the projections module? :) 

One important note is to position variables at the beginning of the methods etc. so it is easier to trail their change through the whole function - you can see all the variables from the beginning. And if there is to much of them, then maybe it is time to add an internal method inside of an existing one.

File: highcharts-api/highcharts/9-3d-charts/main.js [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+
+    data.forEach((p) => {
+        for (i = 0; i <= numberOfPoints; i++) {
+            newShadow.push([
+                0,
+                p.y + ry * cos(i * alpha),
+                p.z + rz * sin(i * alpha),
+            ]);
+        }
+        newShadow.push(null);
```
[the remark this answers is not in the document]
Comment (hubertkozik) [reply highcharts-api/highcharts/9-3d-charts/main.js:90]: To create a space between drawn dots, if there won't be 'nulls', points would be joined together with some paths/areas.

File: highcharts-api/highcharts/9-3d-charts/main.js [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+    });
+    return newShadow;
+};
+
+const addShadowSeries = (data, chart, color) => {
+    chart.addSeries({
+        type: 'polygon',
+        zIndex: 1,
+        color: color,
```
[the remark this answers is not in the document]
Comment (hubertkozik) [reply highcharts-api/highcharts/9-3d-charts/main.js:115]: Yes, it is passed from`chart.events.load` function

File: highcharts-api/highcharts/9-3d-charts/main.js [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+            enabled: true,
+            alpha: 10,
+            beta: 45,
+            depth: 250,
+            viewDistance: 5,
+            fitToPlot: false,
+            frame: {
+                bottom: {
+                    size: 1,
+                    color: 'rgba(0,0,0,1)',
```
[the remark this answers is not in the document]
Comment (hubertkozik) [reply highcharts-api/highcharts/9-3d-charts/main.js:138]: Changed to just 'black'.

File: highcharts-api/highcharts/9-3d-charts/main.js [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+                    }
+                    if (j === maxSteps) {
+                        j = 0;
+                    }
+
+                    // BLUE POINT
+                    blueZIndex =
+                        pointCameraDistance(bluePoint, chart) <
+                        pointCameraDistance(yellowPoint, chart)
+                            ? -1
```
[the remark this answers is not in the document]
Comment (hubertkozik) [reply highcharts-api/highcharts/9-3d-charts/main.js:207]: Yes, it is working.

**Comment by hubertkozik** [comment]
According to the projections module - I didn't find it when I was doing that exercise - it could be much more easier with that module :)

**Review by Izothep — APPROVED** [review]

**Pull request closed without merging**